apps/docs/pages/guides/functions/debugging.mdx

```diff
@@ -11,82 +11,135 @@ export const meta = {
+- **Logs**: shows any platform events, uncaught exceptions, and custom log events. You can see the timestamp, level, and message of each log event. You can also filter the log events by date, time, or level.
 
-### Debugging in production
-
-You can debug your deployed Edge Functions using the ["Functions" section](https://supabase.com/dashboard/project/_/functions) of the Dashboard. There are two debugging tools available:
-
-- Invocations: shows the Request and Response for each execution.
-- Logs: shows any platform events, uncaught exceptions, and custom log events.
+You can access both tools from the [Functions section](https://supabase.com/dashboard/project/_/functions) of the Dashboard. Select your function from the list, and click `Invocations` or `Logs`.
 
 ![Function invocations.](/docs/img/guides/functions/function-logs.png)
 
-### Debugging locally
+### How to access logs locally
 
 When [developing locally](/docs/guides/functions/local-development) you will see error messages and console log statements printed to your local terminal window.
 
-## Limitations
+### How to use logs
 
-- Outgoing connections to ports `25`, `465`, and `587` are not allowed.
-- Serving of HTML content is not supported (`GET` requests that return `text/html` will be rewritten to `text/plain`).
-- Deno FileSystem APIs are not available to use within Edge Functions.
-- Importing modules using `npm:` specifier is currently not supported ([use a CDN to load npm modules](/docs/guides/functions/import-maps)).
+- **Uncaught exceptions**: Uncaught exceptions thrown by a function during execution will be automatically logged, allowing you to view the error message and stack trace in the Logs tool.
+- **Custom log events**: You can also use `console.log`, `console.error`, and `console.warn` in your code to emit custom log events. These events will also appear in the Logs tool. For example, you can use `console.log` to print the value of a variable, or `console.error` to report an unexpected condition.
+- **Message length and event limits**: A custom log message can contain up to 10,000 characters. A function can log up to 100 events within a 10 second period.
 
-## Troubleshooting
+Here is an example of how to use `console.log` in your function:
 
-### Unable to call Edge Function
+```
+Deno.serve(async (_req) => {
+  const { name } = await req.json()
+  const data = {
+    message: `Hello ${name}!`,
+  }
 
-If you're unable to call your Edge Function or are experiencing any CORS issues:
+  console.log(`Name: ${name}`)
 
-- Make sure you followed the [CORS guide](/docs/guides/functions/cors).
-- Check your function logs. Navigate to the [Functions section in your dashboard](https://supabase.com/dashboard/project/_/functions), select your function from the list, and click `Logs`. Check for any errors.
+  return new Response(JSON.stringify(data), { headers: { 'Content-Type': 'application/json' } })
```

Comment:
we can probably extend the example to show all 3 types `log`, `warn`, and `error`.  Code can check if the name is empty and if so add a `console.warn("empty name provided")`. Also, can have a try..catch block and if there's an error in function it can be logged using `console.error`.

Reply:
Is this example better? 

```
Deno.serve(async (req) => {
  try {
    const { name } = await req.json();

    if (!name) {
      console.warn("Empty name provided");
    }

    const data = {
      message: `Hello ${name || 'Guest'}!`, // Provide a default value if name is empty
    };

    console.log(`Name: ${name}`);

    return new Response(JSON.stringify(data), { headers: { 'Content-Type': 'application/json' } });
  } catch (error) {
    console.error(`Error processing request: ${error.message}`);
    return new Response(JSON.stringify({ error: 'Internal Server Error' }), { status: 500, headers: { 'Content-Type': 'application/json' } });
  }
});
```